nav2_regulated_pure_pursuit_controller/test/test_regulated_pp.cpp

```diff
@@ -642,6 +642,105 @@ TEST(RegulatedPurePursuitTest, testDynamicParameter)
   EXPECT_EQ(node->get_parameter("test.use_rotate_to_heading").as_bool(), false);
 }
 
+
+
+TEST(RegulatedPurePursuitTest, extended_collision_check_one_pose)
+{
+  auto node = std::make_shared<rclcpp_lifecycle::LifecycleNode>("testRPP");
+  std::string name = "PathFollower";
+  auto tf = std::make_shared<tf2_ros::Buffer>(node->get_clock());
+  auto costmap = std::make_shared<nav2_costmap_2d::Costmap2DROS>("fake_costmap");
+
+  // instantiate
+  auto ctrl = std::make_shared<BasicAPIRPP>();
+  costmap->on_configure(rclcpp_lifecycle::State());
+  ctrl->configure(node, name, tf, costmap);
+  ctrl->activate();
+
+  // Set up test path;
+  geometry_msgs::msg::PoseStamped start_of_path;
+  start_of_path.header.frame_id = "test_path_frame";
+  start_of_path.header.stamp = node->get_clock()->now();
+  start_of_path.pose.position.x = 0.0;
+  start_of_path.pose.position.y = 0.0;
+  start_of_path.pose.position.z = 0.0;
+
+  constexpr double spacing = 2.0;
+  constexpr double path_length = 2.0;
+
+  auto global_plan = path_utils::generate_path(
+    start_of_path, spacing, {
+    std::make_unique<path_utils::Straight>(path_length)
+  });
+  ctrl->setPlan(global_plan);
+
+  EXPECT_FALSE(ctrl->isCollisionImminentExtendedSearchWrapper());
+
+  ctrl->deactivate();
+  ctrl->cleanup();  
+}
+
+TEST(RegulatedPurePursuitTest, extended_collision_check_cost_map)
+{
+  auto node = std::make_shared<rclcpp_lifecycle::LifecycleNode>("testRPP");
+  std::string name = "PathFollower";
+  auto tf = std::make_shared<tf2_ros::Buffer>(node->get_clock());
+  auto costmap = std::make_shared<nav2_costmap_2d::Costmap2DROS>("fake_costmap");
+
+  // instantiate
+  auto ctrl = std::make_shared<BasicAPIRPP>();
+  costmap->on_configure(rclcpp_lifecycle::State());
+  ctrl->configure(node, name, tf, costmap);
+  ctrl->activate();
+
+  // Set up test path;
+  geometry_msgs::msg::PoseStamped start_of_path;
+  start_of_path.header.frame_id = "test_path_frame";
+  start_of_path.header.stamp = node->get_clock()->now();
+  start_of_path.pose.position.x = 0.0;
+  start_of_path.pose.position.y = 0.0;
+  start_of_path.pose.position.z = 0.0;
+
+  constexpr double spacing = 0.1;
+  constexpr double path_length = 2.0;
+
+  auto global_plan = path_utils::generate_path(
+    start_of_path, spacing, {
+    std::make_unique<path_utils::Straight>(path_length)
+  });
+  ctrl->setPlan(global_plan);
+
+
+  unsigned int width = costmap->getLayeredCostmap()->getCostmap()->getSizeInCellsX();
+  unsigned int height = costmap->getLayeredCostmap()->getCostmap()->getSizeInCellsY();
+
+  // collision should be imminent for inscribed_inflated and lethal but not for free space
+
+  for (unsigned int x = 0; x < width; ++x) {
+      for (unsigned int y = 0; y < height; ++y) {
+          costmap->getLayeredCostmap()->getCostmap()->setCost(x, y, nav2_costmap_2d::LETHAL_OBSTACLE);
```

Comment:
Could be a more meaningful/ strict-er unit test if you could test that a part of the path is in collision and part of it is not. 
So for example, if someone were to change if_any() to all_of() inside `isCollisionImminentExtendedSearch()` these tests would succeed, but we want them to fail.